fractal/parser.py

```diff
@@ -423,8 +431,17 @@
     argument_default=ap.SUPPRESS,
     allow_abbrev=False,
 )
+task_delete_id_or_name_group = task_delete_parser.add_mutually_exclusive_group(
+    required=True
+)
+task_delete_id_or_name_group.add_argument(
+    "--id", help="ID of the task to delete", type=int
+)
+task_delete_id_or_name_group.add_argument(
+    "--name", help="Name of the task to delete", type=str
+)
 task_delete_parser.add_argument(
-    "task_id_or_name", help="ID or name of task to delete", type=str
+    "--version", help="Version of the task to delete", type=str
```

Comment:
Add to version help message something like "(only accepted in combination with --name)"